sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -345,7 +345,7 @@ def visit_transform(self, transform_node):
           for ix, side_input in enumerate(transform_node.side_inputs):
             access_pattern = side_input._side_input_data().access_pattern
             if access_pattern == common_urns.side_inputs.ITERABLE.urn:
-              if use_unified_worker:
+              if use_unified_worker or not use_fn_api:
```

Comment:
I'm a bit confused about the purpose of this change, if `new_side_input` is only used when `use_fn_api` is True?

Reply:
The coder gets updated.

Reply:
So `side_input.pvalue.element_type = typehints.Any` is a side effect and `new_side_input = _DataflowIterableSideInput(side_input)` has no effect?

Reply:
Yes, in non-FnAPI mode.